README.md

```diff
@@ -1,8 +1,7 @@
 # python-observabilityclient
```

Comment:
To maintain consistency everywhere, could we rename this as python-telemetryclient?